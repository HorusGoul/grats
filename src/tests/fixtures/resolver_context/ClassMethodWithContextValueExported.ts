export type GratsContext = {
  greeting: string;
};

/** @gqlType */
export class Query {
  /** @gqlField */
  greeting(args: unknown, ctx: GratsContext): string {
    return ctx.greeting;
  }
}
