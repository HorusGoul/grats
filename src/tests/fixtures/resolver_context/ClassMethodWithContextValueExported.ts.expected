-----------------
INPUT
----------------- 
export type GratsContext = {
  greeting: string;
};

/** @gqlType */
export class Query {
  /** @gqlField */
  greeting(args: unknown, ctx: GratsContext): string {
    return ctx.greeting;
  }
}

-----------------
OUTPUT
-----------------
schema {
  query: Query
}

directive @methodName(name: String!) on FIELD_DEFINITION

directive @exported(filename: String!, functionName: String!) on FIELD_DEFINITION

type Query {
  greeting: String
}